plugin/pkg/auth/authorizer/rbac/rbac.go

```diff
@@ -121,6 +121,8 @@ func (r *RBACAuthorizer) Authorize(requestAttributes authorizer.Attributes) (aut
 	reason := ""
 	if len(ruleCheckingVisitor.errors) > 0 {
 		reason = fmt.Sprintf("RBAC: %v", utilerrors.NewAggregate(ruleCheckingVisitor.errors))
+	} else {
```

Comment:
I don't think is correct.  This changed a very clear message into a less clear message:

```
User “smarterclayton” cannot update namespaces in the namespace “ci-op-r5wf6x70”
```

becomes

```
User “smarterclayton” cannot update namespaces in the namespace “ci-op-r5wf6x70”: no RBAC policy matched
```

I'm a human.  I don't need to know that no RBAC policy matched.  That's actually confusing, because now I think I might be missing a policy.  I don't need a reason when the reason is that I don't have permission.

Reply:
The addition of this clause makes the problem look like an actual bug in the product, not "you lack permission".

Reply:
> This changed a very clear message into a less clear message:
> `User “smarterclayton” cannot update namespaces in the namespace “ci-op-r5wf6x70”`
> becomes
> `User “smarterclayton” cannot update namespaces in the namespace “ci-op-r5wf6x70”: no RBAC policy matched`

the message wasn't clear if any authorizers returned denial reasons. an alternative to this could be to stop returning detailed reasons from authorizers altogether. I'm not sure if that's better or not.